src/models/Scholarship.ts

```diff
@@ -0,0 +1,29 @@
+import scholarships from '../testdata/scholarships';
+
+interface ScholarshipProps {
+  // TODO(https://github.com/beyondhb1079/s4us/issues/56):
+  // Update this to reflect the schema
+  name: string;
+  description: string;
+  deadline: string;
+  website: string;
+  school: string;
+  year: string;
+}
+
+export default class Scholarship {
+  id?: string;
+  data: ScholarshipProps;
+
+  constructor(data: ScholarshipProps) {
+    this.data = { ...data };
+  }
+
+  static get(id: string): Scholarship | undefined {
+    const scholarshipList = scholarships as { [id: string]: ScholarshipProps };
```

Comment:
I was having a little trouble making sense of this but if I understand correctly, the 'as' is telling the compiler to interpret the scholarship as an object with key of type string and value of type ScholarshipProps. 
I'm wondering why the id is wrapped with []

Reply:
Yeah, it's a little confusing to read. I'm not that familiar with TypeScript so I'm not sure if this is the right way to write this but it looks like this is what's called an [Index Type](https://www.typescriptlang.org/docs/handbook/advanced-types.html#index-types-and-index-signatures). TypeScript also has Map and Record types as well though so maybe one of those is better for this case. We can submit this as is though and as we learn more about TypeScript correct this as needed